openstack/identity/identity.go

```diff
@@ -283,16 +277,24 @@ type ValidAdmin struct {
 // This custom handler allows us to more cleanly return an error and response,
 // and pass some package level context into the handler.
 type Handler struct {
-	Client        *gophercloud.ServiceClient
-	Next          http.Handler
-	ValidServices []ValidService
-	ValidAdmins   []ValidAdmin
+	Client         *gophercloud.ServiceClient
+	Next           http.Handler
+	ValidServices  []ValidService
+	ValidAdmins    []ValidAdmin
+	tokenGetResult getResult
 }
 
 // ServeHTTP satisfies the http handler interface.
 // It will check to make sure that the api caller is validated with
 // keystone before allowing the next handler in the chain to be called.
 func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	token := r.Header["X-Auth-Token"]
+	if token != nil {
+		res := v3tokens.Get(h.Client, token[0])
+		h.tokenGetResult = getResult{res}
```

Comment:
Isn't this racy?  Isn't each invocation of ServeHTTP called in a different goroutine?